src/backend/logging/log_buffer.h

```diff
@@ -0,0 +1,72 @@
+/*-------------------------------------------------------------------------
+ *
+ * log_buffer.h
+ * file description
+ *
+ * Copyright(c) 2015, CMU
+ *
+ * /peloton/src/backend/logging/log_buffer.h
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#pragma once
+#include <cstddef>
+#include <cassert>
+#include <memory>
+#include "backend/logging/log_record.h"
+
+namespace peloton {
+namespace logging {
+
+class BackendLogger;
+
+//===--------------------------------------------------------------------===//
+// Log Buffer
+//===--------------------------------------------------------------------===//
+class LogBuffer {
+ public:
+  LogBuffer(BackendLogger *);
+
+  ~LogBuffer(void){};
+
+  // get serialized data field
+  char *GetData();
+
+  // serialize and write a log record to buffer
+  bool WriteRecord(LogRecord *);
+
+  // clean up and reset content
+  void ResetData();
+
+  size_t GetSize();
+
+  void SetSize(size_t size);
+
+  void SetMaxLogId(cid_t new_max) { max_log_id = new_max; }
+
+  cid_t GetMaxLogId() { return max_log_id; }
```

Comment:
It might be better to either put all small functions in .h file for inline or put all functions in .cpp file for separation of declaration and implementation.
You put some small functions in header and some not. Maybe it is better to unify it.